### PR TITLE
fix(radiobuttons): accessibility issue

### LINKED ISSUE #9: RadioButton accessibility is broken
**Describe the bug**
When using Talkback on radiobutton, there is several accessibility issue : 

- We cannot focus on radiobutton when the element is alone (with no text)
- We don't have the status of the current selected radiobutton. It only says "RadioBox - Double tap to activate", but we don't not if the current state of the radiobutton is selected or unselected
- When we double tap to select or unselect a radiobutton, nothing is announce to the user even if something happen on the device.

**To Reproduce**

https://user-images.githubusercontent.com/1913901/168554222-f051534f-284f-4abc-912e-538452e65e72.mp4

**Expected behavior**
We should : 

- [ ] make radiobutton focusable even when no label is associated to it
- [ ] announce radibutton state in stateDescription for example selected or unselected
- [ ] when state change (meaning when the user perform an action on the radiobutton), we should announce the updated states (this should be automatic if we use stateDescription as mention juste here ⬆️ 

**Screenshots**

**Smartphone (please complete the following information):**

- Device: ALL
- OS: ALL
- Version ALL

### PR DESCRIPTION
**Pull request by CorentinStamperDecathlon**
## Changes description 🧑‍💻

- update radiobuttons sample to add states management
- fix accessibility issue

## Context 🤔

Closes #9

## Checklist ✅

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check compose contract convention. It must follow conventions described [here](/CONVENTIONS.md).
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on a phone device/emulator.
- [x] I have tested on a tablet device/emulator.
- [x] I have tested on a large screen device/emulator.
- If it includes design changes, please ask for a review with a core team designer.

## Screenshots 📸

https://user-images.githubusercontent.com/46709142/185627871-f6f41fc6-d676-44f5-ace1-dd13f652ba5d.mp4

### CODE REVIEW

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/812b3a79-9c2e-4bb2-9075-438931432b07/artifact/a5d3ac3628f74a30/p/a9bb8e66f912c02a14040df5161be308

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/e91f35d0-b720-4a04-820d-c455de041d8b/artifact/6414b59df1a3fb79/p/34c3948d075a51c498c379d17427e475

File: radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt [hunk 1]
```diff
@@ -48,29 +50,41 @@ object VitaminRadioButtons {
...
+                    onClick = {
+                        onClick?.invoke()
+                    }
                 )
+                .semantics(mergeDescendants = true) {},
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            RadioButton(
+                selected = selected,
+                onClick = {
```
Comment (GerardPaligot) [inline radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt:78]: Do you really need this callback? The onClick event will be triggered by the `Row`, no?

File: radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt [hunk 2]
```diff
@@ -48,29 +50,41 @@ object VitaminRadioButtons {
...
+                    onClick?.invoke()
+                },
+                enabled = enabled,
+                interactionSource = interactionSource,
+                colors = radioColors,
+                modifier = Modifier.clearAndSetSemantics {}
```
Comment (GerardPaligot) [inline radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt:84]: Curious to know why you had this modifier here?

File: radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt [hunk 3]
```diff
@@ -48,29 +50,41 @@ object VitaminRadioButtons {
...
+                .semantics(mergeDescendants = true) {},
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            RadioButton(
+                selected = selected,
+                onClick = {
+                    onClick?.invoke()
+                },
+                enabled = enabled,
+                interactionSource = interactionSource,
```
Comment (GerardPaligot) [inline radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt:82]: The interaction shouldn't be put on the `Row`?

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

File: radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt (hunk 1 above)
Comment (CorentinStamperDecathlon) [reply radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt:78]: In the implementation of the RadioButton (compose), if the onClick parameter is null, the RadioButton role is not applied. This role allows in particular to define the properties useful for accessibility.

```
// RadioButton.kt (extract)
val selectableModifier =
        if (onClick != null) {
            Modifier.selectable(
                selected = selected,
                onClick = onClick,
                enabled = enabled,
                role = Role.RadioButton,
                interactionSource = interactionSource,
                indication = rememberRipple(
                    bounded = false,
                    radius = RadioButtonRippleRadius
                )
            )
        } else {
            Modifier
        }
```

I will remove the `onClick?.invoke()`

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/632ecd0e-4309-485c-b09f-e5c1222fe9fc/artifact/de0b1abd3a37532a/p/572c79c338f1c7c5e823a2bbcc6b7d39

**Review by GerardPaligot — APPROVED** [review]

**Pull request merged**